packages/api/lambdas/internal-reconciliation-report.js

```diff
@@ -59,82 +61,104 @@ async function internalRecReportForCollections(recReportParams) {
   );
 
   const collectionPgModel = new CollectionPgModel();
-  const knex = await getKnexClient();
-
-  // get collections from database and sort them, since the scan result is not ordered
-  const [updatedAtRangeParams, dbSearchParams] = convertToDBCollectionSearchObject(recReportParams);
-
-  const dbCollectionsSearched = await collectionPgModel.searchWithUpdatedAtRange(
-    knex,
-    dbSearchParams,
-    updatedAtRangeParams
-  );
-
-  // TODO - improve this sort
-  const dbCollectionItems = sortBy(
-    filterDBCollections(dbCollectionsSearched, recReportParams),
-    ['name', 'version']
-  );
-
-  let okCount = 0;
-  const withConflicts = [];
-  let onlyInEs = [];
-  let onlyInDb = [];
-
-  const fieldsIgnored = ['timestamp', 'updatedAt', 'createdAt'];
-  let nextEsItem = await esCollectionsIterator.peek();
-  let nextDbItem = dbCollectionItems.length !== 0
-    ? translatePostgresCollectionToApiCollection(dbCollectionItems[0])
-    : undefined;
-
-  while (nextEsItem && nextDbItem) {
-    const esCollectionId = constructCollectionId(nextEsItem.name, nextEsItem.version);
-    const dbCollectionId = constructCollectionId(nextDbItem.name, nextDbItem.version);
+  const knex = recReportParams.knex || await getKnexClient();
+
```

Comment:
Same comment as for inventory report, we should avoid retry at this level.  Retry at the db query level if needed

Reply:
Yep this is still in progress, definitely plan to do that!